src/nvim/ex_cmds.c

```diff
@@ -3168,7 +3177,13 @@ static buf_T *do_sub(exarg_T *eap, proftime_T timeout)
   linenr_T old_line_count = curbuf->b_ml.ml_line_count;
   char_u *sub_firstline;    // allocated copy of first sub line
   bool endcolumn = false;   // cursor in last column when done
-  MatchedLineVec matched_lines = KV_INITIAL_VALUE;
+  PreviewLines preview_lines = { KV_INITIAL_VALUE, 0};
+  static linenr_T pre_started;
+  static linenr_T pre_ended;
+  static int pre_src_id; // initialized to 0, see
+                         // https://stackoverflow.com/questions/13251083
```

Comment:
could be explicit, and avoid the comment :)

    static int pre_src_id = 0;

Reply:
Usually we put static locals at the very start of a function. I think it would be surprising to see them in the middle of other locals.

Reply:
About that explizit, I'm not sure about the semantics. I don't want it to be reset to 0 every time the function is called, of course. That won't happen?

Reply:
@KillTheMule won't happen. static locals are initialized only once for the life of the process.

Reply:
Which, thinking about it, makes sense! Thanks, I'll change that.